Expands Russians [READY]

Thunder12345 opened the pull request:
Adds a new set of russian simple animal enemies

> Military gear russian: Tougher than regular russians, buckshot shotgun, drops a turtleneck variant, bulletproof armour and a basic pump shotgun with buckshot
> Russian officer: Weaker than regular russians, 9mm machine pistol, calls for backup when threatened, drops his uniform and a stechkin APS

![cyka](https://cloud.githubusercontent.com/assets/5479091/13010779/946fb40c-d19b-11e5-8daf-4377c39f68d6.png)

Adds new guns for russians to use:

> Nagant revolver: 7 shot revolver, suppressable
> Stechkin APS: 9mm burst fire pistol, 12 round magazine

![blyat](https://cloud.githubusercontent.com/assets/5479091/13011614/7ad216b2-d19f-11e5-963d-994c2f8c7347.png)

Mosin russian is now a child of ranged instead of a 50% chance of being given to the standard ranged
Other things added later as I work out exactly what to give the new russians

Features:
- [x] Nagant revolver
- [x] Give nagant revolver to existing ranged russian
- [x] Stechkin APS
- [x] New russian mobs
- [x] Work out other weapons to give to the new russian mobs

Bugs:
- [x] Can't attach suppressor to the nagant, extends to all non /automatic projectile weapons

Sprites:
- [x] Sprites for the nagant revolver with and without a suppressor
- [x] Sprites for the stechkin APS
- [x] Sprites for new russian soldiers

Commit:
Added nagant revolver, and its new ammo type

TrustyGun commented:
cyka bylat

Thunder12345 commented:
Anyone know why the suppressor thing isn't working?

LanCartwright commented:
here try adding an internal silencer instead.

you can't attach a silencer to a revolver in real life. the nagant revolver was silent because of the sealing mechanism with their bullets and their revolvers.

Thunder12345 commented:
No no, the sealing is WHY a suppressor works on it

neersighted commented on code/modules/projectiles/ammunition/ammo_casings.dm:
```diff
@@ -74,6 +74,11 @@
 	projectile_type = /obj/item/projectile/bullet/weakbullet3
 
+/obj/item/ammo_casing/n762
+	desc = "A 7.62x38mmR bullet casing."
```
mmR?

LanCartwright replied:
Technical name 7.62x38mm Nagant Cartridge Type R(evol'ver) [револьвер]

Thunder12345 replied:
mmR is for rimmed cartridges

LanCartwright commented:
The M1895 never had a suppressor. I mean it would be feasible to attach a silencer on hindsight, but I'd say the reason why it's silent is because of how the 7.62x38mmR cartridge worked in the revolver.

You *could* attach a silencer, but it's making me cringe.

Thunder12345 commented:
https://www.youtube.com/watch?v=n97Yrb-OuVY

Commit:
Adds a bunch of parent calls to children of /gun/projectile so they can be suppressed correctly. Nagant revolver now properly accepts suppressors.

Niknakflak commented:
Leave the russian expert stuff to the dirty commies in training.  And raz, an actual russian.

Commit:
Changed ranged russian's gun to nagant revolver
Makes mosin carrying russian a subtype instead of a 50% chance on a normal ranged russian

optimumtact commented on code/modules/projectiles/guns/projectile/saw.dm:
```diff
@@ -55,6 +55,9 @@
 
 /obj/item/weapon/gun/projectile/automatic/l6_saw/attackby(obj/item/A, mob/user, params)
+	. = ..()
+	if(.)
```
what is even going on here?

RemieRichards replied:
if ..() returns 1 we stop, 1 is returned when an automatic weapon gets a "tactical reload"

optimumtact replied:
sweet, thanks

Razharas commented:
# This pr gets an official seal of approval from the russian community of me

![et_punlorbg](https://cloud.githubusercontent.com/assets/5615294/12993696/c8c02f9e-d12c-11e5-8b23-6f1bc2dc0b80.jpg)

Zilenan91 commented:
Any chance of adding a mosin nagant russian or just strictly handguns

optimumtact commented:
Add the obrez

Thunder12345 commented:
>Mosin russian is now a child of ranged instead of a 50% chance of being given to the standard ranged
There's already a mosin variant

Commit:
Added 9mm stechkin APS, 12 round magazine, 3 round burst fire or semi auto.

Commit:
Adds military russian
Adds camouflage fatiges, functionally identical to turtlenecks
Adds buckshot loaded pump shotgun variant
Adds sprites for the new russian

Commit:
Adds russian officer uniform
Adds russian officer
Minor tweaks

Commit:
Oops forgot to make the APS unsuppressable

Thunder12345 commented:
Coderart get

Commit:
Sprites for APS and nagant

Commit:
Added 7.62x38mmR ammo box with 14 rounds

tkdrg commented on code/modules/mob/living/simple_animal/hostile/russian.dm:
```diff
@@ -32,17 +32,40 @@
+/mob/living/simple_animal/hostile/russian/ranged/officer/Aggro()
 	..()
+	summon_backup(15)
+	say("NEED BACKUP!!")
```
It would be nice if they could speak some russian gibberish here. @Razharas your skills are necessary

RemieRichards replied:
It would also have to be in english characters, as russian and other unicode breaks a lot in byond.

tkdrg replied:
It uses your local non-unicode locale but yeah. You can write russian in latin characters though.

RemieRichards replied:
yeah obvi but I just wanted razharas/whoever gives us russian to give it us in latin/english chars

Cheridan replied:
"CHEEKI BREEKI!"

Razharas replied:
"Nuzhno podkreplenie" or "nuzhna podmoga" but both sounds like something lower rank would ask from higher ranks, not otherwise
Logically he should shout "V boj!" as in ATTACK! since he is giving orders to the lower ranks

Commit:
Stechkin buff

KorPhaeron commented on code/modules/mob/living/simple_animal/hostile/russian.dm:
```diff
@@ -32,17 +32,40 @@
+	icon_state = "russianofficer"
+	icon_living = "russianofficer"
+	maxHealth = 65
+	health = 65
```
This health value might be a bit low

Commit:
Cyka blyat

AnturK commented on code/modules/projectiles/guns/projectile.dm:
```diff
@@ -17,6 +17,10 @@
 
+/obj/item/weapon/gun/projectile/update_icon()
+	..()
+	icon_state = "[initial(icon_state)][suppressed ? "-suppressed" : ""]"
```
Why initial here ? Any gun have more than one state ?

neersighted replied:
initial because otherwise `-suppressed` will be appended forever?

AnturK replied:
Oh rightm it has no helper var.

Thunder12345 commented:
**REEEEEEEEEEEEEEEEE**
